numpy/lib/function_base.py

```diff
@@ -3836,8 +3836,10 @@ def _median(a, axis=None, out=None, overwrite_input=False):
         kth = [szh - 1, szh]
     else:
         kth = [(sz - 1) // 2]
-    # Check if the array contains any nan's
-    if np.issubdtype(a.dtype, np.inexact):
+
+    # Check if the array contains any nan's or NaT's (unordered values)
+    supports_nans = np.issubdtype(a.dtype, np.inexact) or a.dtype.kind == 'm'
```

Comment:
if you want to support dt64 here, something like:

```
if a.dtype.kind == "M":
     td_res = _median(a.view("m8"), ...)
     return td_res.view(a.dtype)
```

?

Reply:
Yes, we could.  I won't make it a priority though (i.e. we can follow up if we want these changes and allow median).  We could also replace the call to `mean()` with a calculation that does: `a + (b-a) / 2` (or `*0.5`) for example.